### ISSUE TITLE
Use expressions in attribute values

### ISSUE THREAD

**Issue by cossssmin** [posthtml/posthtml-extend#35]
@Scrum what do you think about adding support for `posthtml-expressions` inside attribute values?

So we could then do:

```html
<extends src="{{ condition ? 'primary' : 'secondary' }}.html">
```

... or:

```html
<block name="{{ condition ? 'foo' : 'bar' }}">
```

I know you can just use `posthtml-expressions` in the `plugins: []` option, but this can result in multiple expression evaluations if your build chain uses `posthtml-expressions` later on ([example](https://github.com/maizzle/framework/blob/adfdcd4ff60500f2eac12097a07bdd36ba3e0d9d/src/generators/posthtml.js#L31-L60)).

**Comment by Scrum** [comment]
@cossssmin I have two opinions on this:
1. I don't really like the logic in templates - it makes the template unreadable
2. I don't really like to search for logic from templates - it takes my time and switches between files

I would prefer to move the logic into the method. the logic can be more complex and it will not be readable in templates

```js
locals: {
  isProduction: (env) => env === 'production' ? 'active' : 'hidden'
}
```
```html
<p class="{{ isProduction(env) }}">in production!</p>
```

**Comment by cossssmin** [comment]
> I would prefer to move the logic into the method

Of course, my examples were very basic, users would be free to choose how they handle it.

I came to think of this since there's no easy way to programatically extend a layout file - right now you have to write the `<extend>` tag twice, which means code duplication that could easily be avoided if we could evaluate simple expressions in the attribute values.

Should I get started on a PR to show how I imagine this working?

**Comment by Scrum** [comment]
> Should I get started on a PR to show how I imagine this working?

I think this will be a good start.
